src/panels/KaitoModelsPanel.ts

```diff
@@ -0,0 +1,295 @@
+                    age: 0,
+                } as InitialState;
+            }
+        }
+        const data = JSON.parse(kubectlresult.result.stdout);
+        const conditions: Array<{ type: string; status: string }> = data.status?.conditions || [];
+        let resourceReady = null;
+        let inferenceReady = null;
+        let workspaceReady = null;
+        conditions.forEach((condition) => {
+            if (condition.type === "ResourceReady") {
+                resourceReady = this.statusToBoolean(condition.status);
+            } else if (condition.type === "WorkspaceReady") {
+                workspaceReady = this.statusToBoolean(condition.status);
+            } else if (condition.type === "InferenceReady") {
+                inferenceReady = this.statusToBoolean(condition.status);
+            }
+        });
+        return {
+            clusterName: this.clusterName,
+            modelName: model,
+            workspaceExists: true,
+            resourceReady: resourceReady,
+            inferenceReady: inferenceReady,
+            workspaceReady: workspaceReady,
+            age: this.convertAgeToMinutes(data.metadata?.creationTimestamp),
+        } as InitialState;
+    }
+
+    private async handleUpdateStateRequest(model: string, webview: MessageSink<ToWebViewMsgDef>) {
+        const command = `get workspace workspace-${model} -o json`;
+        let kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        if (failed(kubectlresult)) {
+            kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            if (failed(kubectlresult)) {
+                vscode.window.showErrorMessage(
+                    `There was an error connecting to the workspace. ${kubectlresult.error}`,
+                );
+                webview.postDeploymentProgressUpdate({
+                    clusterName: this.clusterName,
+                    modelName: "",
+                    workspaceExists: false,
+                    resourceReady: null,
+                    inferenceReady: null,
+                    workspaceReady: null,
+                    age: 0,
+                });
+                return;
+            }
+        }
+        const data = JSON.parse(kubectlresult.result.stdout);
+        const conditions: Array<{ type: string; status: string }> = data.status?.conditions || [];
+        let resourceReady = null;
+        let inferenceReady = null;
+        let workspaceReady = null;
+
+        conditions.forEach((condition) => {
+            if (condition.type === "ResourceReady") {
+                resourceReady = this.statusToBoolean(condition.status);
+            } else if (condition.type === "WorkspaceReady") {
+                workspaceReady = this.statusToBoolean(condition.status);
+            } else if (condition.type === "InferenceReady") {
+                inferenceReady = this.statusToBoolean(condition.status);
+            }
+        });
```

Comment:
💡 :suggestion and observation: How about `switch` statement for clarity and **destructures** the condition object directly in the loop, making it more concise and readable? 

I think below will just work out of box.

```*.ts
let resourceReady = null;
let inferenceReady = null;
let workspaceReady = null;

conditions.forEach(({ type, status }) => {
    switch (type) {
        case "ResourceReady":
            resourceReady = this.statusToBoolean(status);
            break;
        case "WorkspaceReady":
            workspaceReady = this.statusToBoolean(status);
            break;
        case "InferenceReady":
            inferenceReady = this.statusToBoolean(status);
            break;
    }
});
```